src/gems/simulation/simulation_table.py

```diff
@@ -1,118 +1,104 @@
-        self,
-        output_values: OutputValues,
-        block: int,
-        absolute_time_offset: int,
-        block_size: int,
-        scenario_count: int,
-    ) -> None:
-        """
-        Populate the SimulationTable from solver output values and append the objective value.
-        """
-        rows = []
+class SimulationColumns(str, Enum):
+    BLOCK = "block"
+    COMPONENT = "component"
+    OUTPUT = "output"
+    ABSOLUTE_TIME_INDEX = "absolute-time-index"
+    BLOCK_TIME_INDEX = "block-time-index"
+    SCENARIO_INDEX = "scenario-index"
+    VALUE = "value"
+    BASIS_STATUS = "basis-status"
 
-        # Loop over components in OutputValues
-        for comp_id, comp_obj in output_values._components.items():
-            # Loop over variables for the component
-            for var_name, var in comp_obj._variables.items():
-                size_s, size_t = var._size  # (scenario_count, time_count)
-
-                # Fallback if sizes don't match provided scenario_count/block_size
-                if size_s == 0:
-                    size_s = scenario_count
-                if size_t == 0:
-                    size_t = block_size
-
-                for s in range(size_s):
-                    for t in range(size_t):
-                        value = var._value.get(TimeScenarioIndex(t, s), None)
-                        basis_status = var._basis_status
-
-                        rows.append(
-                            {
-                                "simulation_id": self.simulation_id,
-                                "block": block,
-                                "component": comp_id,
-                                "output": var._name,
-                                "absolute_time_index": absolute_time_offset + t,
-                                "block_time_index": t + 1,
-                                "scenario_index": s,
-                                "value": value,
-                                "basis_status": basis_status,
-                            }
-                        )
-
-        # Store results as DataFrame
-        self.df = pd.DataFrame(rows)
-
-        # Ensure problem is set before accessing solver
+
+class SimulationTableBuilder:
+    """Builds simulation tables from solver output values."""
+
+    def __init__(self, simulation_id: Optional[str] = None) -> None:
+        self.simulation_id = simulation_id or datetime.now().strftime("%Y%m%d-%H%M")
+
+    def build(self, output_values: OutputValues) -> pd.DataFrame:
+        """Populate a DataFrame from OutputValues."""
         if output_values.problem is None:
             raise ValueError("OutputValues problem is not set.")
-        objective_value = output_values.problem.solver.Objective().Value()
 
-        # Append the objective value as the last row
+        context = output_values.problem.context
+        block = context._block.id
+        block_size = context.block_length()
+        absolute_time_offset = (block - 1) * block_size
```

Comment:
This assumes that all blocks are the same size... this may not be the case in the most general case

Reply:
This is why I think this is ok to have this as argument